k8s-operator/sessionrecording/hijacker.go

```diff
@@ -147,6 +148,7 @@ func (h *Hijacker) setUpRecording(ctx context.Context, conn net.Conn) (net.Conn,
 		errChan <-chan error
 	)
 	h.log.Infof("kubectl %s session will be recorded, recorders: %v, fail open policy: %t", h.sessionType, h.addrs, h.failOpen)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Do we want to add any kind of deadline here? Not sure what the potential for blocking forever would be.

Reply:
yeah so the only issue with a timeout here is that, how long will the kubectl session be? An exec session could go on for many hours right? And from what I understand we need this context to stay open for the duration.